api/query/search.go

```diff
@@ -141,6 +137,54 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
+	// Check if the query is a simple link query.
```

Comment:
As discussed offline: This was intended for the case that no test runs are passed, only a metadata search.

I don't think we should be handling this case server-side; a client that is fetching only metadata, without respect to any particular runs, will be in a different context than one that is calling the `/api/search` endpoint (and can simply call `/api/metadata` directly).